src/coreclr/jit/morph.cpp

```diff
@@ -13550,6 +13545,26 @@ GenTree* Compiler::fgOptimizeEqualityComparisonWithConst(GenTreeOp* cmp)
             DEBUG_DESTROY_NODE(rshiftOp->gtGetOp2());
             DEBUG_DESTROY_NODE(rshiftOp);
         }
+
+        // Optimizes (X & 1) == 1 to (X & 1)
+        // 
+        //                        EQ/NE                   AND
+        //                        /  \                   /  \.
+        //                      AND   CNS 1  ->         x   CNS 1
+        //                     /   \                  
+        //                    x   CNS 1
+        //           
+        // GTF_RELOP_JMP_USED is used to make sure the optimization is used for return statements only.
+        if (op1->OperIs(GT_AND) && !(cmp->gtFlags & GTF_RELOP_JMP_USED))
+        {
+            if (op1->gtGetOp2()->IsIntegralConst(1))
+            {
+                DEBUG_DESTROY_NODE(cmp->gtGetOp2());
+                DEBUG_DESTROY_NODE(cmp);
+
+               return op1;
```

Comment:
Please update the function header: `Currently only returns relop trees.` -> `Currently only returns GTK_SMPOP trees.`